app/models/api_request.rb

```diff
@@ -0,0 +1,47 @@
+class ApiRequest
+  def query(query)
+    uri = request_uri(query)
```

Comment:
if you take my above suggestion this can become 

```ruby
def response
  res = send_api_request(request_uri)
  format_response(res)
end
```
(i would do that all in one line and not create the var but that's my personal style :))